pkg/app/api/grpcapi/piped_api.go

```diff
@@ -968,12 +968,50 @@ func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pip
 			return nil, err
 		}
 	}
+
+	opts := datastore.ListOptions{
+		Filters: []datastore.ListFilter{
+			{
+				Field:    "ProjectId",
+				Operator: datastore.OperatorEqual,
+				Value:    projectID,
+			},
+			{
+				Field:    "Deleted",
+				Operator: datastore.OperatorEqual,
+				Value:    false,
+			},
+		},
+	}
+	envs, err := a.environmentStore.ListEnvironments(ctx, opts)
+	if err != nil {
+		a.logger.Error("failed to get environments", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to get environments")
+	}
+	envMap := make(map[string]string, len(envs))
+	for _, env := range envs {
+		// It is assumed that the env name is kept unique by the user.
+		envMap[env.Name] = env.Id
+	}
+
 	for _, appInfo := range req.Applications {
+		envID := ""
+		if appInfo.EnvName != "" {
+			var ok bool
+			envID, ok = envMap[appInfo.EnvName]
+			if !ok {
+				a.logger.Error("unknown environment name given",
+					zap.String("env", appInfo.EnvName),
+					zap.Error(err),
+				)
+				return nil, status.Errorf(codes.InvalidArgument, "Unknown environment name %q given", appInfo.EnvName)
+			}
```

Comment:
How about ignoring this error since users can specify any new environment value?

Reply:
Right. if already registered env is given, it should update the Envrironment as well as the label. Otherwise, it just updates the label.